input/system/selfhosted/otel_handler.go

```diff
@@ -105,73 +108,103 @@ func skipDueToK8sFilter(kubernetes *common.KeyValueList, server *state.Server, p
 	logParser := server.GetLogParser()
-	go func() {
-		http.HandleFunc("/v1/logs", func(w http.ResponseWriter, r *http.Request) {
-			b, err := io.ReadAll(r.Body)
-			if err != nil {
-				prefixedLogger.PrintError("Could not read otel body")
-			}
-			logsData := &otlpLogs.LogsData{}
-			if err := proto.Unmarshal(b, logsData); err != nil {
-				prefixedLogger.PrintError("Could not unmarshal otel body")
-			}
-			for _, r := range logsData.ResourceLogs {
-				for _, s := range r.ScopeLogs {
-					for _, l := range s.LogRecords {
-						var logger string
-						var record *common.KeyValueList
-						var kubernetes *common.KeyValueList
-						hasErrorSeverity := false
-						if l.Body.GetKvlistValue() != nil {
-							for _, v := range l.Body.GetKvlistValue().Values {
-								if v.Key == "logger" {
-									logger = v.Value.GetStringValue()
-								}
-								if v.Key == "record" {
-									record = v.Value.GetKvlistValue()
-								}
-								if v.Key == "kubernetes" {
-									kubernetes = v.Value.GetKvlistValue()
-								}
-								if v.Key == "error_severity" {
-									hasErrorSeverity = true
-								}
-							}
-							// TODO: Support other logger names (this is only tested with CNPG)
-							if logger == "postgres" {
-								// jsonlog wrapped in K8s context (via fluentbit)
-								logLine, detailLine := logLineFromJsonlog(record, logParser)
-								if skipDueToK8sFilter(kubernetes, server, prefixedLogger) {
-									continue
-								}
-
-								parsedLogStream <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: logLine}
-								if detailLine != nil {
-									parsedLogStream <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: *detailLine}
-								}
-							} else if logger == "" && hasErrorSeverity {
-								// simple jsonlog (Postgres jsonlog has error_severity key)
-								logLine, detailLine := logLineFromJsonlog(l.Body.GetKvlistValue(), logParser)
-								parsedLogStream <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: logLine}
-								if detailLine != nil {
-									parsedLogStream <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: *detailLine}
-								}
-							}
-						} else if l.Body.GetStringValue() != "" {
-							// plain log message
-							item := SelfHostedLogStreamItem{}
-							item.Line = l.Body.GetStringValue()
-							item.OccurredAt = time.Unix(0, int64(l.TimeUnixNano))
-							rawLogStream <- item
+	b, err := io.ReadAll(r.Body)
+
+	if err != nil {
+		prefixedLogger.PrintError("Could not read otel body")
+	}
+	logsData := &otlpLogs.LogsData{}
+	if err := proto.Unmarshal(b, logsData); err != nil {
+		prefixedLogger.PrintError("Could not unmarshal otel body")
+	}
+
+	if opts.DebugLogs {
```

Comment:
Just talked about this a bit with @keiko713 and others in PR review.

It may make sense to introduce a new "verbose verbose" (`-vv`) mode in the collector that emits this information, whilst also emitting snapshots. The current `--debug-logs` flag isn't that useful for customer environments since you would have to stop the collector, start it with that flag on, and when the flag is on no snapshots get sent (i.e. customer would experience it as missing data).